certs/certs.go

```diff
@@ -264,6 +280,14 @@ func certsToRemove(oldCerts, newCerts []*x509.Certificate) map[string]*x509.Cert
 		newCertMap[certID] = struct{}{}
 	}
 
+	// We don't want to "rotate" certificates to an empty set, nor keep old certificates if the
+	// new root does not trust them.  newCerts should come from validRootLeafCerts, which refuses
```

Comment:
It might be worth bubbling up a mention of this comment to the function doc comment above, so this method isn't called in other contexts